modules/social_features/social_tagging/src/SocialTaggingService.php

```diff
@@ -174,7 +202,7 @@ public function buildHierarchy(array $terms, $entity_type) {
       // Get current terms parents.
       $parents = $this->termStorage->loadParents($current_term->id());
       $parent = reset($parents);
-      $category = $parent->getName();
+      $category = $this->getTranslatedLabel($parent->id());
```

Comment:
While the function getTranslatedLabel receives tid and then it loads the term inside that function, however, so far, in the instances where this function is called, terms have been already loaded, so can we improve this function to just accepting taxonomy terms as an entity and then returning the correct language version?